src/main/java/core/basesyntax/service/operations/SupplyOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void getOperation(List<FruitTransaction> transactions) {
+        List<FruitTransaction> transactionsSelected = transactions.stream()
+                .filter(a -> a.getOperation() == FruitTransaction.Operation.SUPPLY)
+                .toList();

```

Comment:
Filtering the transactions should not be the responsibility of the `SupplyOperation` class. The list of transactions passed to this method should already be filtered by the operation type. This logic should be handled at a higher level, such as in the strategy implementation or the main service logic.